doc/sphinx/proofs/writing-proofs/equality.rst

```diff
@@ -31,44 +31,64 @@ There are multiple notions of :gdef:`equality` in Coq:
   other advantages.  ":term:`Convertible <convertible>`" is another way of saying that
   two terms are definitionally equal.
 
-Equality
---------
-
-.. tacn:: f_equal
-
-   For a goal with the form :n:`f a__1 ... a__n = g b__1 ... b__n`, creates
-   subgoals :n:`f = g` and :n:`a__i = b__i` for the `n` arguments. Subgoals
-   that can be proven by :tacn:`reflexivity` or :tacn:`congruence` are solved
-   automatically.
+Reflexivity, symmetry and transitivity tactics
+----------------------------------------------
 
 .. tacn:: reflexivity
 
-   For a goal with the form `t=u`, verifies that `t` and `u` are :term:`convertible`,
-   and if so,
+   For a goal with the form `t = u`, verifies that `t` and `u` are
+   :term:`definitionally equal <definitional equality>`, and if so,
    solves the goal.  If not, it fails.  It is equivalent to ``apply refl_equal``.
 
-   .. exn:: The conclusion is not a substitutive equation.
-      :undocumented:
+   The tactic may also be applied to goals with the form `R @term__1 @term__2` where
+   `R` is an equivalence relation registered with the `Equivalence` or `Reflexive`
+   typeclasses.  See :cmd:`Class` and :cmd:`Instance`.
+
+   .. todo PR: an "equivalence relation" is not necessarily a "setoid equality", right?
 
-   .. exn:: Unable to unify ... with ...
+   .. todo PR: I suppose the form of the goal is really
+      :n:`{? forall @open_binders , } EQ @term__1 @term__2` as described in
+      :tacn:`rewrite`.  Worth repeating here or is that just tedious?
```

Comment:
Indeed, you're right. You could include the `{? forall @open_binders , }` part without really talking much about it.

Reply:
Got it.  Slightly distracting to see this boilerplate multiple times here but it would be odd to see it only once in the following text.  WDYT?

![image](https://user-images.githubusercontent.com/1253341/115583956-39406580-a27f-11eb-9252-59cb407d5177.png)

Reply:
Actually, this description is not correct. What it does is that it introduces the variables from the open binders in the context.

Reply:
OK, I can keep the first forall here and not mention the detail that the binder variables are introduced into the context.  Similar treatment for the other two.